### PR TITLE
[FLINK-17340][docs] Update docs which related to default planner changing

### PR DESCRIPTION
**Pull request by KurtYoung**
## What is the purpose of the change

Update the documents.

### CODE REVIEW

File: docs/dev/table/catalogs.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ Set a `JdbcCatalog` with the following parameters:
 <div data-lang="Java" markdown="1">
 {% highlight java %}
 
-EnvironmentSettings settings = EnvironmentSettings.newInstance().useBlinkPlanner().inStreamingMode().build();
+EnvironmentSettings settings = EnvironmentSettings.newInstance().inStreamingMode().build();
```
Comment (libenchao) [inline docs/dev/table/catalogs.md:66]: also update for `hive/index.md` ?

File: docs/dev/table/common.md [hunk 2]
```diff
@@ -32,12 +32,11 @@ Main Differences Between the Two Planners
...
-7. The Blink planner will optimize multiple-sinks into one DAG (supported only on `TableEnvironment`, not on `StreamTableEnvironment`). The old planner will always optimize each sink into a new DAG, where all DAGs are independent of each other.
-8. The old planner does not support catalog statistics now, while the Blink planner does.
+3. The implementations of `FilterableTableSource` for the old planner and the Blink planner are incompatible. The old planner will push down `PlannerExpression`s into `FilterableTableSource`, while the Blink planner will push down `Expression`s.
+4. String based key-value config options (Please see the documentation about [Configuration]({{ site.baseurl }}/dev/table/config.html) for details) are only used for the Blink planner.
+5. The implementation(`CalciteConfig`) of `PlannerConfig` in two planners is different.
+6. The Blink planner will optimize multiple-sinks into one DAG (supported only on `TableEnvironment`, not on `StreamTableEnvironment`). The old planner will always optimize each sink into a new DAG, where all DAGs are independent of each other.
+7. The old planner does not support catalog statistics now, while the Blink planner does.
```
Comment (libenchao) [inline docs/dev/table/common.md:39]: https://ci.apache.org/projects/flink/flink-docs-master/dev/table/common.html#translate-and-execute-a-query
https://ci.apache.org/projects/flink/flink-docs-master/dev/table/common.html#query-optimization
These two places have tabs 'blink' and 'old planner', do you think we can put 'blink' section above 'old planner' section, then blink will be shown to the user by default (no need to have one more click).

Comment (KurtYoung) [reply docs/dev/table/common.md:39]: yes, nice catch.

**Comment by KurtYoung** [comment]
@libenchao thanks for your review, I've addressed your comment.

**Comment by libenchao** [comment]
@KurtYoung Thanks for addressing my comments, the changes LGTM now.

**Pull request closed without merging**